concordium-contracts-common/src/types.rs

```diff
@@ -41,6 +45,21 @@ pub struct Amount {
     pub micro_ccd: u64,
 }
 
+#[cfg(feature = "concordium-quickcheck")]
+impl quickcheck::Arbitrary for Amount {
+    fn arbitrary(g: &mut Gen) -> Amount {
+        Amount {
+            micro_ccd: quickcheck::Arbitrary::arbitrary(g),
+        }
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        Box::new(quickcheck::Arbitrary::shrink(&self.micro_ccd).map(|mccd| Amount {
+            micro_ccd: mccd,
+        }))
```

Comment:
This should also work, and I think it's nicer
```suggestion
        Box::new(quickcheck::Arbitrary::shrink(&self.micro_ccd).map(Amount::from_micro_ccd))
```